feat: prettier UI

nmattela opened the pull request.

lpalinckx commented:
Very clean UI - Olga would approve 

Some comments:
- After selecting a subtitle, the subtitle list:
1. scrolls back to the top of the list, can this be prevented? 
2. resets the search result (all the other results are gone and you have to re-search for them)
- Can the "Generate GIF" button and "Caption" textfield also fit in the view for a desktop browser? Having to scroll to a button is exhausting :( 
- When selecting "all videos" the GIF settings dissapear, but when selecting any  specifc episode from the list, they do appear. Is this intended? Shouldn't the settings be visible at all times?
- I don't know if i am a fan of the current 'menu' for selecting individual episodes. When there are a lot of results, it seems a bit small to use it with a scrollbar, maybe a dropdown like on mobile instead? It is fine when there aren't any (or not a lot of) results for subtitles when there's a bit more space. 
- There's no feedback for the user if an error occurred during the request for a GIF (e.g. font size is set too large), perhaps a little snackbar will help.

Mobile: 
- Maybe add a way to get back to the subtitle list after getting to the GIF settings 
- Perhaps some padding (see screenshot) 
![screenshot_2025-01-28_12-09-12](https://github.com/user-attachments/assets/2cbb04ea-d5b5-4e36-b292-4285930b0033)

Overall great job! Blijven hutsen!

nmattela commented:
> Very clean UI - Olga would approve
Thonx :3
> 
> Some comments:
> 
>     * After selecting a subtitle, the subtitle list:
> 
> 
>     1. scrolls back to the top of the list, can this be prevented?
I've had to make a couple of changes to make mobile view possible the way it is now. One of those changes is to re-render the entire content of the screen (except for header and footer) instead of just a specific part as it was before. This is because in mobile, you have to hide the subtitle list while showing the settings form, thus both parts need a re-render. This re-render is the same for mobile as for desktop so that's why the state resets like that. I'll look for solutions to this, perhaps having the gif settings pop up as a modal, so that only the modal needs a re-render and it will work intuitively for both mobile and desktop.
> 
>     2. resets the search result (all the other results are gone and you have to re-search for them)
That's strange, I'll look into that. The search bar is normally exempt from the re-render discussed above.
> 
> 
>     * Can the "Generate GIF" button and "Caption" textfield also fit in the view for a desktop browser? Having to scroll to a button is exhausting :(
I've probably overlooked that while developing the mobile view, will have a look
> 
>     * When selecting "all videos" the GIF settings dissapear, but when selecting any  specifc episode from the list, they do appear. Is this intended? Shouldn't the settings be visible at all times?
Maybe this is some caching issue. Will have a look
> 
>     * I don't know if i am a fan of the current 'menu' for selecting individual episodes. When there are a lot of results, it seems a bit small to use it with a scrollbar, maybe a dropdown like on mobile instead? It is fine when there aren't any (or not a lot of) results for subtitles when there's a bit more space.
You're right. It will be easier to have the same controls on desktop as on mobile. I will go for a dropdown
> 
>     * There's no feedback for the user if an error occurred during the request for a GIF (e.g. font size is set too large), perhaps a little snackbar will help.
Yes, I was intending on implementing that still
> 
> 
> Mobile:
> 
>     * Maybe add a way to get back to the subtitle list after getting to the GIF settings
Yes, was planning on doing that
> 
>     * Perhaps some padding (see screenshot)
>       ![screenshot_2025-01-28_12-09-12](https://private-user-images.githubusercontent.com/45123100/407288956-2cbb04ea-d5b5-4e36-b292-4285930b0033.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzgwODM5NTYsIm5iZiI6MTczODA4MzY1NiwicGF0aCI6Ii80NTEyMzEwMC80MDcyODg5NTYtMmNiYjA0ZWEtZDViNS00ZTM2LWIyOTItNDI4NTkzMGIwMDMzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTI4VDE3MDA1NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI1MjNmMDk0YWJlNzQ0MWM3YWU1MWNmYzkxM2JhYjcxOTIxYjBkOTNkMzgzMmI0YjMwNmQyNmM0ZmU5NDk2N2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.5a6cnaGtZOBysZfOHEfr07gG5OkClSTW0JtWZL6dWRs)
> 
> 
> Overall great job! Blijven hutsen!

Niboor commented on README.md:
````diff
@@ -4,6 +4,42 @@ An HTTP server that can generate GIFs on demand, based on subtitles in video fil
 
+## Running the application
+
+Install Tailwind and Daisy UI (CSS packages):
+```
+yarn
+```
+
+Whenever making changes to the CSS, rerun the tailwind script:
+```
+npx @tailwindcss/cli -i public/main.css -o public/tailwind.css
+```
+
+Start a virtual environment:
+```
+python -m venv .
+```
````
Do we really need to pull in the whole JS ecosystem in order to compile some CSS? Is there a better way? Since we're using htmx I was hoping we could avoid lots of the build complexity that comes with modern frontend.

nmattela replied:
There's actually no JavaScript involved that goes to the client-side, except HTMX and the little JS code in main.js. This JavaScript is actually used to compile the tailwind styles. Tailwind also offers a CDN option, but they state that it should only be used for development purposes, so I refrained from using that

Niboor commented on package.json:
_If_ the JS package mess is unavoidable, could we put that in a separate directory (with the rest of the frontend stuff)?

nmattela replied:
I wouldn't do that tbh, otherwise you will need to start messing with calling commands in a subdirectory and moving files out of it into the public dir

Niboor reviewed (commented):
@lpalinckx Already gave some feedback on the style (I'll wait until you've made some more changes). Here's some comments that came up while skimming through the code.

lpalinckx commented:
- After generating a GIF, closing the modal and clicking again on a subtitle, the placeholder text says "The GIF is loading..."
- Is there a reason to put a dropdown inside a dropdown? Include => All Videos => <episode list>
- Wasn't the modal supposed to prevent the scrolling back to top after closing the modal? Because it still happens 
- after searching any subtitle, the include dropdown changes to the very first episode, but still searches for in all files

nmattela commented:
> * After generating a GIF, closing the modal and clicking again on a subtitle, the placeholder text says "The GIF is loading..."
Fixed
>     * Is there a reason to put a dropdown inside a dropdown? Include => All Videos => <episode list>
Did this to support multiple series in the future. For now it looks a bit wonky, true. Would you prefer it gone?
>     * Wasn't the modal supposed to prevent the scrolling back to top after closing the modal? Because it still happens. In my opinion, if this issue can't be easily resolved, I prefer the previous layout with the subs on the left and the GIF settings permanently visible on the right for desktop browsers.
Fixed
>     * after searching any subtitle, the include dropdown changes to the very first episode, but still searches for occurrences in all files
Fixed

Niboor commented:
Thanks for all the work!
